skipruntime-ts/ts/src/internals/skipruntime_module.ts

```diff
@@ -628,9 +628,9 @@ class NonEmptyIteratorImpl<T> implements NonEmptyIterator<T> {
     );
 
     return {
-      next() {
+      next(): IteratorResult<T> {
         const value = cloned_iter.next();
-        return { value, done: value == null } as IteratorResult<T>;
+        return value === null ? { value, done: true } : { value };
```

Comment:
Does it make a difference that `done` is not explicitly set (to `false`) in the else case?

Reply:
No, it works as well.